### PR TITLE
crdgen: handle OIDCConnectorSpecV3.MaxAge as a special case

### PR DESCRIPTION
**Pull request by capnspacehook**
Attempting to make a nullable protobuf field with a duration type that can also have its resource's YAML intuitively modified has proven to be a challenge. [Another PR that introduces `OIDCConnectorSpecV3.MaxAge`](https://github.com/gravitational/teleport/pull/29815) broke the CRD generated schema. I've tried updating the code to handle this case in a general sense but am not sure it's worth the effort, hence this workaround suggested by @tigrato.

For context in Slack on how I arrived at the current version of `MaxAge` and discussion on a fix for `crdgen` see [here](https://gravitational.slack.com/archives/C0DF0TPMY/p1690918861065649) and [here](https://gravitational.slack.com/archives/C0DF0TPMY/p1690934661564389).

### CODE REVIEW

**Review by hugoShaka — CHANGES_REQUESTED** [review]
I understand you don't want to revamp the whole logic to handle a special case. Could you group all the special case logic in a specific function and keep the main logic clean? This way, if we need to add another special field, this will remain readable.

for example
```go
if jsonName == "" {
    err = handleNamelessFields(schema, message, field)
    if err != nil {
        return nil, trace.WrapWithMessage(err, "empty json tag for %s.%s", message.Name(), field.Name())
    }
    continue
}

// ...

func handleNamelessFields(schema, message, field) error {
    // All the custom logic you need, set properties on the schema here, return an error if the field is not a special case
}
```

**Comment by capnspacehook** [comment]
I updated the testdata, using the branch from https://github.com/gravitational/teleport/pull/29815 so that these changes can be tested. That PR is approved and `OIDCConnectorSpecV3.MaxAge` will not change, so I think it's safe to do so.

**Review by hugoShaka — APPROVED** [review]

File: integrations/operator/crdgen/testdata/protofiles/teleport/accesslist/v1/accesslist.proto [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.accesslist.v1;
```
Comment (tigrato) [inline integrations/operator/crdgen/testdata/protofiles/teleport/accesslist/v1/accesslist.proto:17]: You don't need to copy all these files.

Comment (hugoShaka) [reply integrations/operator/crdgen/testdata/protofiles/teleport/accesslist/v1/accesslist.proto:17]: In a near future I think I'll just remove the testdata and use the actual proto files from `api/proto`, updating them is cumbersome and we end up with XXL PRs for nothing

Comment (capnspacehook) [reply integrations/operator/crdgen/testdata/protofiles/teleport/accesslist/v1/accesslist.proto:17]: I just ran `make update-protos`, but I'll go ahead and delete all unnecessary files

**Review by tigrato — APPROVED** [review]

**Review by tigrato — APPROVED** [review]
Approved again - we raced

**Review by r0mant — APPROVED** [review]

**Pull request merged**